client/packages/common/src/ui/layout/tables/components/Cells/NumberInputCell/NumberInputCell.tsx

```diff
@@ -47,6 +47,7 @@ export const NumberInputCell = <T extends RecordWithId>({
       }}
       onChange={num => {
         const newValue = num === undefined ? min : num;
+        if (buffer === newValue) return;
         setBuffer(newValue);
         updater({ ...rowData, [column.key]: Number(newValue) });
```

Comment:
This is what resolves the issue of the allocation warnings being hidden when the pack quantity input is only focused/defocused (rather than actually changed in any way.) `updater` ends up triggering allocation, which clears the warning...

In #3814 I took the opposite approach, expecting `updater` to handle being called with the same value multiple times. However, I don't really see any value in calling these state updates if the value hasn't changed in the first place?